lib/recipient.rb

```diff
@@ -0,0 +1,67 @@
+require "httparty"
+require "pry"
+
+require "dotenv"
+Dotenv.load
+
+module SlackCli
+  class SlackError < StandardError; end
+
+  class Recipient
+    MSG_URL = "https://slack.com/api/chat.postMessage"
+    TOKEN = ENV["SLACK_TOKEN"]
+
+    attr_reader :send_message, :name, :slack_id, :error_helper, :details, :list
+    # :nocov:
+    def initialize
+      @slack_id = slack_id
+      @name = name
+    end
+    # :nocov:
+
+    def self.list_url
+      raise NotImplementedError, "TODO: implement me in a child class"
+    end
+
+    def post_message(name, message)
+      body_params = {
+        token: TOKEN,
+        as_user: true,
+        channel: name,
+        text: message,
+      }
+      response = HTTParty.post(MSG_URL, body: body_params)
+
+      if response != 200 && !response["ok"]
+        raise SlackCli::SlackError "Error: #{response["error"]}"
+      else
+        return response
+      end
+    end
+
+    def self.get
+      query_params = {
+        token: TOKEN,
+      }
+      response = HTTParty.get(list_url, query: query_params)
+      return error_helper(response)
+    end
+    
+    # :nocov:
+    def details
+      raise NotImplementedError "TODO: implement me in a child class"
+    end
```

Comment:
Instead of excluding this from coverage, why not have a test that `Recipient` by itself without a subclass will raise this exception when this method is called?